core/query/src/main/java/datawave/core/query/configuration/QueryData.java

```diff
@@ -0,0 +1,576 @@
+        }
+        exists = in.readBoolean();
+        if (exists) {
+            lastResult = new Key();
+            lastResult.readFields(in);
+        }
+        finished = in.readBoolean();
+    }
+
+    public QueryData(ObjectInput in) throws IOException {
+        readExternal(in);
+    }
+
+    @Override
+    public int hashCode() {
+        if (rebuildHashCode) {
+            //  @formatter:off
+            hashCode = new HashCodeBuilder()
+                    .append(tableName)
+                    .append(query)
+                    .append(ranges)
+                    .append(columnFamilies)
+                    .append(settings)
+                    .append(lastResult)
+                    .append(finished)
+                    .hashCode();
+            rebuildHashCode = false;
+            //  @formatter:on
+        }
+        return hashCode;
+    }
+
+    public boolean equals(Object o) {
+        if (o instanceof QueryData) {
+            QueryData other = (QueryData) o;
+            //  @formatter:off
+            return new EqualsBuilder()
+                    .append(tableName, other.tableName)
+                    .append(query, other.query)
+                    .append(ranges, other.ranges)
+                    .append(columnFamilies, other.columnFamilies)
+                    .append(settings, other.settings)
+                    .append(lastResult, other.lastResult)
+                    .append(finished, other.finished)
+                    .isEquals();
+            //  @formatter:on
+        }
+        return false;
+    }
+
+    /**
+     * Method to reset the hashcode when an internal variable is updated
+     */
+    private void resetHashCode() {
+        rebuildHashCode = true;
+    }
+
+    /**
+     * A json deserializer for a list of IteratorSetting which handles the json deserialization issues. The accumulo IteratorSetting does not have a default
+     * constructor.
+     */
+    public static class IteratorSettingListDeserializer extends StdDeserializer<List<IteratorSetting>> {
+        private ObjectMapper mapper = new ObjectMapper();
+
+        public IteratorSettingListDeserializer() {
+            this(null);
+        }
+
+        public IteratorSettingListDeserializer(Class<?> valueClass) {
+            super(valueClass);
+        }
+
+        @Override
+        public List<IteratorSetting> deserialize(JsonParser parser, DeserializationContext deserializer) throws IOException, JsonProcessingException {
+            List<IteratorSetting> list = new ArrayList<>();
+            ObjectCodec codec = parser.getCodec();
+            JsonNode node = codec.readTree(parser);
+
+            for (int i = 0; i < node.size(); i++) {
+                list.add(getIteratorSetting(node.get(i)));
+            }
+
+            return list;
+        }
+
+        private IteratorSetting getIteratorSetting(JsonNode node) throws JsonProcessingException {
+            IteratorSetting setting = new IteratorSetting(1, "a", "a");
```

Comment:
this initialization is a bit strange without a comment

Reply:
reworked to avoid hokey constructor params